mappings/net/minecraft/nbt/AbstractNbtNumber.mapping

```diff
@@ -0,0 +1,33 @@
+CLASS net/minecraft/class_2514 net/minecraft/nbt/AbstractNbtNumber
+	COMMENT Represents an NBT number.
+	COMMENT <p>
+	COMMENT The type value {@code 99} is an alias for
+	COMMENT {@linkplain NbtCompound#contains checking the existence of a number element in a NBT compound object}.
+	METHOD method_10696 getShort ()S
+		COMMENT Gets the value as a 16-bit integer.
+		COMMENT
+		COMMENT @return the value as a short
+	METHOD method_10697 getDouble ()D
+		COMMENT Gets the value as a 64-bit floating-point number.
+		COMMENT
+		COMMENT @return the value as a double
+	METHOD method_10698 getByte ()B
+		COMMENT Gets the value as an 8-bit integer.
+		COMMENT
+		COMMENT @return the value as a byte
+	METHOD method_10699 getLong ()J
+		COMMENT Gets the value as a 64-bit integer.
+		COMMENT
+		COMMENT @return the value as a long
+	METHOD method_10700 getFloat ()F
+		COMMENT Gets the value as a 32-bit floating-point number.
+		COMMENT
+		COMMENT @return the value as a float
+	METHOD method_10701 getInt ()I
+		COMMENT Gets the value as a 32-bit integer.
+		COMMENT
+		COMMENT @return the value as an int
+	METHOD method_10702 getNumber ()Ljava/lang/Number;
```

Comment:
Is renaming this to `getNumeric` a good choice?

Reply:
Feels a bit weird.

Reply:
Maybe `getValue`? `number.getNumber()` feels a bit weird compared to `number.getValue()`. Also we have type-specific getInt/Double etc. and this shouldn't be too confusing

Reply:
What about `getGeneric`?

Reply:
`value`/`numberValue`, `doubleValue`, `intValue`, etc. Akin to JDK's `Number` class

Reply:
JDK's Number is immutable, NBT's isn't.

Reply:
Except NBT's is, all the number type definitions (or reader before) override a method which checks if the type is immutable and return `true` for the number types.

Reply:
we shouldn't expect immutability. it was added in sone snapshot, and they can always change nbt behavior as they see fit.

Reply:
You are correct, it even switched to constructor init only lately. Adopting the Number conventions may be reasonable then. `numberValue` above `Value` though, don't want to encourage using it.

Reply:
liach they were semi-immutable before with a package private deserializer, now they are fully so. Doesn't look like that'd change.

Reply:
Agreed. *If* we decide to use `xxValue` names, we should change other value getters from `getXxx` to `xxxValue`, including `getInt` to `intValue`, `getNumber` to `numberValue`, etc.